corehq/apps/reports/standard/sms.py

```diff
@@ -677,7 +679,28 @@ def get_filters(self):
                 else:
                     content_type_filter.append(content_type)
 
-        return (source_filter, content_type_filter)
+        event_status = EventStatusFilter.get_value(self.request, self.domain)
+        if event_status == MessagingEvent.STATUS_COMPLETED:
+            event_status_filter = Q(status=event_status)
+        elif event_status == MessagingEvent.STATUS_ERROR:
+            event_status_filter = (
+                Q(status=event_status) |
+                Q(messagingsubevent__status=event_status) |
+                Q(messagingsubevent__sms__error=True)
+            )
+        elif event_status == MessagingEvent.STATUS_IN_PROGRESS:
+            event_status_filter = (
+                Q(status=event_status) |
+                Q(messagingsubevent__status=event_status) |
+                Q(messagingsubevent__xforms_session__end_time__isnull=True)
+            )
+        elif event_status == MessagingEvent.STATUS_NOT_COMPLETED:
+            event_status_filter = (
+                Q(messagingsubevent__xforms_session__end_time__isnull=False) &
+                Q(messagingsubevent__xforms_session__submission_id__isnull=True)
+            )
+
+        return source_filter, content_type_filter, event_status_filter
```

Comment:
yep, looks good, just we should remove STATUS_COMPLETED from the options and instead put an "All" option in its place. So that way, either the filter does nothing or filters for one of: errors, in progress, not completed.